hudi-common/src/main/java/org/apache/hudi/common/util/ExternalFilePathUtil.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.common.util;
+
+/**
+ * Utility methods for handling externally created files.
+ */
+public class ExternalFilePathUtil {
+  // Suffix acts as a marker when appended to a file path that the path was created by an external system and not a Hudi writer.
+  private static final String EXTERNAL_FILE_SUFFIX = "_hudiext";
```

Comment:
oh, we are changing it to suffix is it. whats the rationale?

Reply:
For ease of appending this to the path in the write stats that are passed in the commit. Now we go from something like  `partition1/file_1.parquet` to `partition1/file_1.parquet_20230805150812_hudiext` instead of  `partition1/hudi_ext_20230805150812_file_1.parquet`.

Reply:
gotcha